mypyc/irbuild/format_str_tokenizer.py

```diff
@@ -0,0 +1,63 @@
+"""Tokenizers for three string formatting methods"""
+
+import re
+from typing import List, Tuple
+
+from mypyc.ir.ops import Value, Integer
+from mypyc.ir.rtypes import c_pyssize_t_rprimitive
+from mypyc.irbuild.builder import IRBuilder
+from mypyc.primitives.str_ops import str_build_op
+
+
+def tokenizer_printf_style(format_str: str) -> Tuple[List[str], List[str]]:
+    # printf-style String Formatting:
+    # https://docs.python.org/3/library/stdtypes.html#old-string-formatting
+    pattern = re.compile(r"""
+        (
+        %                                # Start sign
+        (?:\((?P<key>[^)]*)\))?          # Optional: Mapping key
+        (?P<flag>[-+#0 ]+)?              # Optional: Conversion flags
+        (?P<width>\d+|\*)?               # Optional: Minimum field width
+        (?:\.(?P<precision>\d+|\*))?     # Optional: Precision
+        [hlL]?                           # Optional: Length modifier, Ignored
+        (?P<type>[diouxXeEfFgGcrsa])     # Conversion type
+        | %%)
+        """, re.VERBOSE)
```

Comment:
Make this pattern a top-level constant, otherwise it will be compiled on every function call.